addons/spreadsheet/static/src/pivot/odoo_pivot.js

```diff
@@ -202,6 +203,10 @@ export class OdooPivot {
             const { dimensionWithGranularity, isPositional, field } =
                 this.parseGroupField(nameWithGranularity);
             if (isPositional) {
+                const table = this.getTableStructure();
+                // @ts-ignore: That's a VERY ugly hack to make sure the table is computed and sorted (which is needed for
```

Comment:
If its necessary then the data structure is wrong. At the very least, this should be extracted in another method that can be overridden by the child class?
To be discussed irl, i think

Reply:
Actually; the child class overrides getTableStructure so you should theoretically receive the sorted table? Or is there something I'm missing?

Reply:
Indeed I can just call `this.getTableStructure()` and the table will be sorted the call of `this.sortTableStructure(table);` is not needed but I wanted to add it to be more epxlicit about what we're doing.

The problem is that I need to call `this.getTableStructure()` here because we _know_ that it will be overwritten by the child (the presentation layer) and the table will be sorted. If we don't do it, the pivot_model will call it's own this.getTableStructure(), will not sort the table, and will return the wrong values for the positional pivot functions.

I talked about this a bit with pro and lul, and we ended up at the conclusion that we're fucked. But we should probably talk about it more.